pkg/provider/azure_loadbalancer.go

```diff
@@ -3069,13 +3069,108 @@ func (az *Cloud) reconcileSecurityRules(sg network.SecurityGroup,
 
 	updatedRules = removeDuplicatedSecurityRules(updatedRules)
 
+	if dirtySg || shouldTidySecurityRules(updatedRules) {
+		updatedRules = tidySecurityRules(updatedRules)
+		dirtySg = true
+	}
+
 	for _, r := range updatedRules {
 		klog.V(10).Infof("Updated security rule while processing %s: %s:%s -> %s:%s", service.Name, logSafe(r.SourceAddressPrefix), logSafe(r.SourcePortRange), logSafeCollection(r.DestinationAddressPrefix, r.DestinationAddressPrefixes), logSafe(r.DestinationPortRange))
 	}
 
 	return dirtySg, updatedRules, nil
 }
 
+// shouldTidySecurityRules returns true if the priorities of rules should be re-ordered.
+// The rules should be tidied if there are any allow rules after deny rules which will not work.
+func shouldTidySecurityRules(rules []network.SecurityRule) bool {
+	if len(rules) <= 1 {
+		return false
+	}
+	sort.Slice(rules, func(i, j int) bool {
+		return *rules[i].Priority < *rules[j].Priority
```

Comment:
What if rules to be created do not have Priority yet?

Reply:
It should and will only be called after the priority is assigned to it. I assume the assignment is in https://github.com/kubernetes-sigs/cloud-provider-azure/blob/2a6e42fd6d00071509f4d506c7ba884b1b252795/pkg/provider/azure_loadbalancer.go#L3064

Reply:
oh I see. Thanks for explanation.